src/compile.ts

```diff
@@ -66,7 +68,7 @@ export async function compile(workdir: string, options: Options) {
   if (options.python) {
     targets.python = {
       distName: 'generated',
-      module: options.python.moduleName,
+      module: options.python.moduleName.replace(/-/g, '_'),
```

Comment:
I am assuming java would have a similar issue. I wonder if the correct behavior here is to validate per the language constraints and throw an error so the user will be required to pass in a valid name for the module.

Reply:
Yeah! We can do some validation instead of just correcting it. I'd be worried in some cases though where the user isn't the one explicitly choosing what this value is. In some use-cases, like CRDs in cdk8s from the internet, they might not control the group.

Would we then add this correction behavior to all consumers?

Reply:
Maybe we can do a `warn` that lets the user know we're correcting it?

Reply:
Ok! I will add validation here then